### ISSUE TITLE
cannot overfitt

### ISSUE THREAD

**Issue by sdimantsd** [jinfagang/yolov7#29]
Hi,

What is the name of the mask loss? I did not see it in the logs.

I am try to overfitt one image. but after the training finished the results are not good.
conf (It's the "yolomask.yaml" config. I changed only the dataset and batch size):

```
_BASE_: "../Base-YOLOv7.yaml"
MODEL:
  META_ARCHITECTURE: "YOLOMask"
  WEIGHTS: ""
  MASK_ON: True
  BACKBONE:
    NAME: "build_cspdarknetx_backbone"
  DARKNET:
    WEIGHTS: ""
    DEPTH_WISE: False
    OUT_FEATURES: ["dark2", "dark3", "dark4", "dark5"]

  YOLO:
    ANCHORS:
      # yolomask anchors slightly different than YOLOv7
      [
        [142, 110],
        [192, 243],
        [459, 401],

        [36, 75],
        [76, 55],
        [72, 146],
        
        [12, 16],
        [19, 36],
        [40, 28],
      ]
    ANCHOR_MASK: [[0, 1, 2], [3, 4, 5], [6, 7, 8]]
    CLASSES: 80
    IN_FEATURES: ["dark2", "dark3", "dark4", "dark5"] # p3, p4, p5 from FPN
    WIDTH_MUL: 0.50
    DEPTH_MUL: 0.33

    CONF_THRESHOLD: 0.001
    NMS_THRESHOLD: 0.65
    IGNORE_THRESHOLD: 0.7
    VARIANT: "yolov7"
    LOSS_TYPE: "v7"
    LOSS:
      LAMBDA_IOU: 1.1
    NECK:
      TYPE: "fpn"
      WITH_SPP: true

DATASETS:
  TRAIN: ("one_image_overfit",)
  TEST:  ("one_image_overfit",)

INPUT:
  MASK_FORMAT: "bitmask"
  MIN_SIZE_TRAIN: (416, 512, 608)
  MAX_SIZE_TRAIN: 608 # force max size train to 800?
  MIN_SIZE_TEST: 416
  MAX_SIZE_TEST: 608
  # open all augmentations
  RANDOM_FLIP: "none"
  JITTER_CROP:
    ENABLED: False
  RESIZE:
    ENABLED: False
    # SHAPE: (540, 960)
  DISTORTION:
    ENABLED: False
  # MOSAIC:
  #   ENABLED: True
  #   NUM_IMAGES: 4
  #   DEBUG_VIS: True
  #   # MOSAIC_WIDTH: 960
  #   # MOSAIC_HEIGHT: 540
  MOSAIC_AND_MIXUP:
    ENABLED: False
    DEBUG_VIS: False
    ENABLE_MIXUP: False

SOLVER:
  # AMP:
  # ENABLED: true
  IMS_PER_BATCH: 8 # 1/5 bs than YOLOX
  # it can be 0.016 maybe
  BASE_LR: 0.0009
  STEPS: (60000, 80000)
  WARMUP_FACTOR: 0.00033333
  WARMUP_ITERS: 1500
  MAX_ITER: 190000
  LR_SCHEDULER_NAME: "WarmupCosineLR"

TEST:
  # EVAL_PERIOD: 10000
  EVAL_PERIOD: 0
OUTPUT_DIR: "output/coco_yolomask"

DATALOADER:
  # proposals are part of the dataset_dicts, and take a lot of RAM
  NUM_WORKERS: 1

```

Also I add this line to register the dataset:
`    
def load_data(json_path):
    with open(json_path, 'r') as file:
        json_dataset = json.load(file)
    return json_dataset

def get_dataset(json_path, dataset_name, classes):

    # DatasetCatalog.register(dataset_name, lambda d=json_path: load_data(d))
    register_coco_instances(dataset_name, {}, json_path, '')
    # MetadataCatalog.get(dataset_name).set(thing_classes=classes)  # ["Dog", "Cat", "Mouse"]

    metadata = MetadataCatalog.get("train")
    return metadata

get_dataset('/home/ws/data/dataset/coco_one_image_overfit/train_coco.json',
                                        'coco_one_image_overfit', ['car', 'bus', 'truck'])`

![0_2](https://user-images.githubusercontent.com/29533264/167294374-becbf953-8049-4ba4-af59-ae6cd0d3cf2c.jpg)

**Comment by jinfagang** [comment]
Dude, even I don't recognize yolo can do few-sample learning? The label you draw should not be your classes name, clearly the mask learns failed. You probably take a look on anchor settings first. As I emphasized in README, everything single dataset applied to anchor based method, you should check your anchors first.

However, YOLOv7 provides YOLOX and transformers options as well. If you get updates, just update here.

**Comment by sdimantsd** [comment]
Hi @jinfagang , Thanks for the answer.
I didn't sure I understand you.
I realized I needed to look at the anchors. But what you mean about the label and about the few-sample learning?

If I need a masks for the objects, what is the best configure for it?

**Comment by jinfagang** [comment]
Sorry, please use it in a normal usage.

**Comment by sdimantsd** [comment]
@jinfagang Sorry, What do you mean "normal usage". Did yoloMask didn't do masks?
